tasks/distribution/restic_auto_restore.yml

```
@@ -0,0 +1,25 @@
+# - name: auto_restore check
+#   debug:
+#     msg: "auto_restore in? {{ restic_backups_loop }}"
+
+# - name: auto_restore enabled
+#   debug:
+#     msg: "auto_restore enabled {{ restic_backups_loop }}"
+#   when: restic_backups_loop.auto_restore is defined
+
+- name: "check for a state file {{ restic_auto_restore_dir }}/{{ restic_backups_loop.name }}.state"
+  stat:
+    path: "{{ restic_auto_restore_dir }}/{{ restic_backups_loop.name }}.state"
+  register: restic_restore_state_file
+  when: restic_backups_loop.auto_restore is defined
```

I think this would fire even if `auto_restore: false`
```suggestion
  when: 
    - restic_backups_loop.auto_restore is defined
    - restic_backups_loop
```